### PR TITLE
checks out of bound time

### PR DESCRIPTION
**Pull request by Ferrariic**
Raises an error for players with out of bound times.

### CODE REVIEW

**Comment by Ferrariic** [comment]
WAIT DON'T APPROVE THAT LMAO

File: api/routers/legacy_debug.py [hunk 1]
```diff
@@ -124,7 +125,21 @@ async def detect(detections:List[detection], manual_detect:int) -> None:
     # data validation, there can only be one reporter, and it is unrealistic to send more then 5k reports.
     if len(df) > 5000 or df["reporter"].nunique() > 1:
         logger.debug('Too many reports.')
-        return {'NOK': 'NOK'}, 400
+        return {'ERROR': 'ERROR'}, 400
```
Comment (ThorntonMatthewD) [inline api/routers/legacy_debug.py:128]: I don't actually think we use this line anymore besides the return itself. I like having ERROR next to it in all-caps though. XD

Comment (Ferrariic) [reply api/routers/legacy_debug.py:128]: lmaoo yeah

File: api/routers/legacy_debug.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ async def detect(detections:List[detection], manual_detect:int) -> None:
     
     utc_upper = (utc + dt.timedelta(hours=time_buffer))
     utc_lower = (utc - dt.timedelta(hours=time_buffer))
-    df_temp = pd.to_datetime(df.timestamp)
+    df_temp = pd.to_datetime(df.ts)
```
Comment (Ferrariic) [inline api/routers/legacy_debug.py:137]: good catch :D

**Review by ThorntonMatthewD — CHANGES_REQUESTED** [review]
The only thing I see that is an issue at the moment is that the exception you raise in detect won't actually get to the user. They will always get a 200 from us: https://github.com/Bot-detector/Bot-Detector-Core-Files/blob/001f6f5f25fb3f8abd5c1cabd24819ad8148ce13/api/routers/legacy_debug.py#L193

We throw the detection processing onto another thread so we don't hold the response hostage (and result in a 500 timeout error on the client side). Anything that happens in `detect` happens after the user has already been given a thumbs up. If you want to give feedback to the user then it'll have to be done before the `asyncio.create_task()` call. It could get kinda weird ngl

**Comment by Ferrariic** [comment]
Tbh that's a fair point. Do we want to silently just not log those queries? It would also help for db spams.

**Review by ThorntonMatthewD — APPROVED** [review]

**Pull request merged**